simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/handlers/ocpp/SignCertificateRequestHandler.java

```diff
@@ -0,0 +1,66 @@
+package com.evbox.everon.ocpp.simulator.station.handlers.ocpp;
+
+import com.evbox.everon.ocpp.simulator.station.StationHardwareData;
+import com.evbox.everon.ocpp.simulator.station.StationMessageSender;
+import com.evbox.everon.ocpp.simulator.station.StationStore;
+import lombok.extern.slf4j.Slf4j;
+import org.bouncycastle.openssl.jcajce.JcaPEMWriter;
+import org.bouncycastle.operator.ContentSigner;
+import org.bouncycastle.operator.OperatorCreationException;
+import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder;
+import org.bouncycastle.pkcs.PKCS10CertificationRequest;
+import org.bouncycastle.pkcs.PKCS10CertificationRequestBuilder;
+import org.bouncycastle.pkcs.jcajce.JcaPKCS10CertificationRequestBuilder;
+
+import javax.security.auth.x500.X500Principal;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.security.*;
+
+@Slf4j
+public class SignCertificateRequestHandler implements Runnable {
+
+    private static final String SIGNATURE_ALG = "SHA256withRSA";
```

Comment:
Honestly I think it would be a bit more fancy if we stick to elliptical certificates in simulator, at least they are shorter and possibly would even fit into 800 bytes. I am not sure whether it makes real sense to make it configurable RSA/ECDSA. Possibly some variable in SecurityCtrl with ECDSA as default.